index.js

```diff
@@ -51,12 +51,12 @@ function fastifyCompress (fastify, opts, next) {
   // add onSend hook onto each route as needed
   fastify.addHook('onRoute', (routeOptions) => {
     // If route config.compress has been set it takes precedence over compress
-    if (routeOptions.config && typeof routeOptions.config.compress !== 'undefined') {
+    if (routeOptions.config?.compress !== undefined) {
       routeOptions.compress = routeOptions.config.compress
     }
 
     // Manage compression options
-    if (typeof routeOptions.compress !== 'undefined') {
+    if (routeOptions.compress !== undefined) {
```

Comment:
This should probably just check routeOptions, right? As that's what the old `typeof` check was doing.